doc/Networking_Userland.md

```diff
@@ -0,0 +1,266 @@
+Tock Userland Networking Design Document
+========================================
+
+_NOTE: This document is a work in progress._
+
+This document describes the current userland interface for the networking stack
+on Tock. This document should serve as a description of the abstraction
+provided by libTock - what the exact system call interface looks like or how
+libTock or the kernel implements this functionality is out-of-scope of this
+document.
+
+TODO: Authors/contributors
+
+## Overview
+The Tock networking stack and libTock should attempt to expose a networking
+interface that is similar to the POSIX networking interface. The primary
+motivation for this design choice is that application programmers are used
+to the POSIX networking interface design, and significant amounts of code
+have already been written for POSIX-style network interfaces. By designing
+the libTock networking interface to be as similar to POSIX as possible, we
+hope to improve developer experience while enabling the easy transition of
+networking code to Tock.
+
+## Design
+#### Author: Hudson Ayers
+#### Last Updated: 8-1-2018
+
+udp.c and udp.h in libtock-c/libtock define the userland interface to the
+Tock networking stack. These files interact with capsules/src/net/udp/driver.rs
+in the main tock repository. driver.rs implements an interface for sending
+and receiving UDP messages. It also exposes a list of interace addresses to
+the application layer. The primary functionality embedded in the UDP driver
+is within the allow(), subscribe(), and command() calls which can be made to
+the driver.
+
+allow() is used to setup buffers to read/qrite from. This function takes in
+an `allow_num` and a slice. These allow\_nums determine which buffer is being
+setup as follows:
```

Comment:
Can you move this to a new `.md` file in the syscalls folder to match the other syscall docs?